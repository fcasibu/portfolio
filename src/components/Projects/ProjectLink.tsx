import * as React from 'react';

interface Props {
  link: string;
  children: React.ReactNode;
}

export const ProjectLink = ({ link, children }: Props) => {
  return (
    <li>
      <a href={link} target="_blank" rel="noreferrer noopener">
        {children}
      </a>
    </li>
  );
};
