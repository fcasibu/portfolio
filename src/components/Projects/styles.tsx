import styled from 'styled-components';

export const ProjectsWrapper = styled.div`
  @media (max-width: 640px) {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0 1rem;
  }
`;

export const ProjectsTitle = styled.h2`
  margin-bottom: 1rem;
  text-align: left;
`;

export const StyledProjectsList = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 7rem;
  grid-row-gap: 2rem;
  justify-content: center;

  @media (max-width: 640px) {
    grid-template-columns: 1fr;
  }
`;

export const StyledProjectCard = styled.div<{ isVisible: boolean }>`
  position: relative;
  width: 250px;
  height: 280px;
  display: flex;
  flex-direction: column;
  gap: 1rem;
  opacity: ${(props) => (props.isVisible ? 1 : 0)};
  transform: ${(props) => (props.isVisible ? 'none' : 'translateY(30px)')};
  visibility: ${(props) => (props.isVisible ? 'visible' : 'hidden')};
  transition: all 600ms ease-in-out;
  will-change: opacity, visibility;

  @media (max-width: 640px) {
    width: 350px;
    height: 330px;
  }

  @media (max-width: 375px) {
    width: 100%;
  }
`;

export const ProjectDescription = styled.p`
  font-size: 0.9rem;
`;

export const ProjectImageWrapper = styled.div`
  position: relative;
  cursor: pointer;

  &:after {
    content: '${(props) => props.title}';
    position: absolute;
    top: 0;
    display: grid;
    place-items: center;
    background: rgba(0, 0, 0, 0.3);
    color: white;
    font-size: 1.2rem;
    border-radius: 0.5rem;
    height: 100%;
    width: 100%;
    opacity: 0;
    visibility: hidden;
    transition: opacity 100ms ease-in-out;
  }

  &:hover:after {
    opacity: 1;
    visibility: visible;
  }

  @media (max-width: 640px) {
    width: 100%;
  }
`;

export const ProjectImage = styled.img`
  height: 150px;
  width: 250px;
  border-radius: 0.5rem;

  @media (max-width: 640px) {
    width: 100%;
    height: 200px;
  }
`;

export const StyledProjectLinks = styled.ul`
  position: absolute;
  bottom: -5px;
  right: 0;
  display: flex;
  gap: 0.3rem;

  display: flex;

  > * {
    font-size: 1.5rem;

    @media (max-width: 640px) {
      font-size: 2rem;
    }
  }
`;
