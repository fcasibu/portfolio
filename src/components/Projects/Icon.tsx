import { IconImage, StyledIcon } from './styles';

interface Props {
  image: string;
}

export const Icon = ({ image }: Props) => {
  return (
    <StyledIcon>
      <IconImage src={image} alt="" />
    </StyledIcon>
  );
};
