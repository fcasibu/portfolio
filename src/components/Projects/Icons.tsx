import { Icon, LinksIcons, StyledIcons } from './styles'
import { FiEye, FiCode } from 'react-icons/fi'

interface Props {
  technologies: NodeRequire[]
}

const renderIcons = (technologies: any[]) => {
  const list = []

  for (let i = 0; i < technologies.length; i++) {
    list.push(<Icon src={technologies[i].default} />)
  }

  return list
}

export const Icons = ({ technologies }: Props) => {
  return (
    <StyledIcons>
      {renderIcons(technologies)}
      <LinksIcons>
        <a>
          <FiEye />
        </a>
        <a>
          <FiCode />
        </a>
      </LinksIcons>
    </StyledIcons>
  )
}
