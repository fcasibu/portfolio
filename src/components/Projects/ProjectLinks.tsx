import { FiCode, FiEye } from 'react-icons/fi';
import { ProjectLink } from './ProjectLink';
import { StyledProjectLinks } from './styles';

interface Props {
  repo: string;
  live: string;
}

export const ProjectLinks = ({ repo, live }: Props) => {
  return (
    <StyledProjectLinks>
      <ProjectLink link={live}><FiEye /></ProjectLink>
      <ProjectLink link={repo}><FiCode /></ProjectLink>
    </StyledProjectLinks>
  );
};
