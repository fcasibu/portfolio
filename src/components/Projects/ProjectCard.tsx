import * as React from 'react';
import { Icons } from './Icons';
import { ProjectLinks } from './ProjectLinks';
import { ProjectDescription, ProjectImage, ProjectImageWrapper, StyledProjectCard } from './styles';

interface Data {
  title: string;
  description: string;
  repo: string;
  live: string;
  image: string;
  technologies: NodeRequire[];
}

interface Props {
  data: Data;
}

export const ProjectCard = ({ data }: Props) => {
  const cardRef = React.useRef<HTMLDivElement>(null)
  const [isVisible, setIsVisible] = React.useState(true);

  React.useEffect(() => {
    const observer = new IntersectionObserver(entries => {
      entries.forEach(entry => setIsVisible(entry.isIntersecting));
    })
    observer.observe(cardRef.current as HTMLDivElement);
    return () => observer.unobserve(cardRef.current as HTMLDivElement);
  }, [])

  return (
    <StyledProjectCard ref={cardRef} isVisible={isVisible}>
      <ProjectImageWrapper title={data.title}>
        <ProjectImage src={data.image} />
      </ProjectImageWrapper>
      <ProjectDescription>{data.description}</ProjectDescription>
      <Icons technologies={data.technologies} />
      <ProjectLinks repo={data.repo} live={data.live} />
    </StyledProjectCard>
  );
};
