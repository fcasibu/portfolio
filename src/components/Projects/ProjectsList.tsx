import { StyledProjectsList } from './styles';
import { projects } from '../../data/projects';
import { ProjectCard } from './ProjectCard';

export const ProjectsList = () => {
  return (
    <StyledProjectsList>
      {projects.map((el) => (
        <ProjectCard key={el.title} data={el} />
      ))}
    </StyledProjectsList>
  );
};
