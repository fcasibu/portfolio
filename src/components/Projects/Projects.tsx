import { ProjectsList } from './ProjectsList';
import { ProjectsTitle, ProjectsWrapper } from './styles';

const Projects = () => {
  return (
    <ProjectsWrapper>
      <ProjectsTitle>Projects</ProjectsTitle>
      <ProjectsList />
    </ProjectsWrapper>
  )
}

export default Projects;
