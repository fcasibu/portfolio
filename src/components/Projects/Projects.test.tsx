import { render, screen, within } from '@testing-library/react';
import '@testing-library/jest-dom';
import Projects from './Projects';
import { ProjectCard } from './ProjectCard';
import { projects } from '../../data/projects';
import { Icons } from './Icons';

describe('Projects', () => {
  beforeEach(() => {
    const mockFn = jest.fn();
    mockFn.mockReturnValue({
      observe: () => null,
      unobserve: () => null
    });
    window.IntersectionObserver = mockFn;
  });

  it('should render and match the snapshot', () => {
    const { container } = render(<Projects />);

    expect(container).toMatchSnapshot();
  });

  it('should render ProjectCard properly', () => {
    render(<ProjectCard data={projects[0]} />);

    const description = screen.getByText(
      /single player battleship game, where the player and computer battles out who sinks the most ships first/i
    );

    expect(description).toBeInTheDocument();
  });

  it('a tags should have the correct link', () => {
    render(<ProjectCard data={projects[0]} />);

    const links = screen.getAllByRole('link') as HTMLLinkElement[];

    expect(links[0].href).toEqual('https://nevz9.github.io/battleship/');
    expect(links[1].href).toEqual('https://github.com/nevz9/battleship');
  });

  it('should display the correct length of tech stack', () => {
    render(<Icons technologies={projects[1].technologies} />);
    const list = screen.getByRole('list');
    const icons = within(list).getAllByRole('listitem');

    expect(icons).toHaveLength(3);
  });
});
