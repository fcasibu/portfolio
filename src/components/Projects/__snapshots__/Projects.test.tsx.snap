// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should render and match the snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ bZmHnT"
  >
    <h2
      class="sc-gsnTZi gZLFFb"
    >
      Projects
    </h2>
    <div
      class="sc-dkzDqf cNHQkR"
    >
      <div
        class="sc-hKMtZM wgLOy"
      >
        <div
          class="sc-jSMfEi eNckjs"
          title="Battleship"
        >
          <img
            class="sc-gKXOVf fmLhIk"
            src="battleship.png"
          />
        </div>
        <p
          class="sc-eCYdqJ hNHXnO"
        >
          Single player battleship game, where the player and computer battles out who sinks the most ships first
        </p>
        <ul
          class="sc-ftvSup fwQdZr"
        >
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="html.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="css.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="javascript.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="jest.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="webpack.svg"
            />
          </li>
        </ul>
        <ul
          class="sc-iBkjds gIKFUO"
        >
          <li>
            <a
              href="https://nevz9.github.io/battleship/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://github.com/nevz9/battleship"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="16 18 22 12 16 6"
                />
                <polyline
                  points="8 6 2 12 8 18"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="sc-hKMtZM wgLOy"
      >
        <div
          class="sc-jSMfEi XpJbA"
          title="Fakegram"
        >
          <img
            class="sc-gKXOVf fmLhIk"
            src="fakegram.png"
          />
        </div>
        <p
          class="sc-eCYdqJ hNHXnO"
        >
          Instagram clone, with features including posting, following, liking, authentication, etc
        </p>
        <ul
          class="sc-ftvSup fwQdZr"
        >
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="react.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="firebase.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="css.svg"
            />
          </li>
        </ul>
        <ul
          class="sc-iBkjds gIKFUO"
        >
          <li>
            <a
              href="https://nevz9.github.io/fakegram-react/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://github.com/nevz9/fakegram-react"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="16 18 22 12 16 6"
                />
                <polyline
                  points="8 6 2 12 8 18"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="sc-hKMtZM wgLOy"
      >
        <div
          class="sc-jSMfEi bCobrj"
          title="Photo Tagging Application"
        >
          <img
            class="sc-gKXOVf fmLhIk"
            src="photo-tagging.png"
          />
        </div>
        <p
          class="sc-eCYdqJ hNHXnO"
        >
          Photo Tagging Application. Three maps. Three Characters. Find all three as fast as you can!
        </p>
        <ul
          class="sc-ftvSup fwQdZr"
        >
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="react.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="firebase.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="jest.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="css.svg"
            />
          </li>
        </ul>
        <ul
          class="sc-iBkjds gIKFUO"
        >
          <li>
            <a
              href="https://photo-tagging-app-852b2.web.app/#/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://github.com/nevz9/photo-tagging-app"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="16 18 22 12 16 6"
                />
                <polyline
                  points="8 6 2 12 8 18"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="sc-hKMtZM wgLOy"
      >
        <div
          class="sc-jSMfEi hmZHtY"
          title="Shopping Cart"
        >
          <img
            class="sc-gKXOVf fmLhIk"
            src="shopping-cart.png"
          />
        </div>
        <p
          class="sc-eCYdqJ hNHXnO"
        >
          Shopping Cart. First time implementing React Router and imitating the logic of shoping and adding to a cart
        </p>
        <ul
          class="sc-ftvSup fwQdZr"
        >
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="react.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="css.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="jest.svg"
            />
          </li>
        </ul>
        <ul
          class="sc-iBkjds gIKFUO"
        >
          <li>
            <a
              href="https://nevz9.github.io/shopping-cart/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://github.com/nevz9/shopping-cart"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="16 18 22 12 16 6"
                />
                <polyline
                  points="8 6 2 12 8 18"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="sc-hKMtZM wgLOy"
      >
        <div
          class="sc-jSMfEi dJMWvB"
          title="Todo App"
        >
          <img
            class="sc-gKXOVf fmLhIk"
            src="todo.png"
          />
        </div>
        <p
          class="sc-eCYdqJ hNHXnO"
        >
          Todo App. First big project to deal with CRUD operations and using local storage for data persistence
        </p>
        <ul
          class="sc-ftvSup fwQdZr"
        >
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="html.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="css.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="javascript.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="webpack.svg"
            />
          </li>
        </ul>
        <ul
          class="sc-iBkjds gIKFUO"
        >
          <li>
            <a
              href="https://nevz9.github.io/todo-list/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://github.com/nevz9/todo-list"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="16 18 22 12 16 6"
                />
                <polyline
                  points="8 6 2 12 8 18"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="sc-hKMtZM wgLOy"
      >
        <div
          class="sc-jSMfEi eOazwe"
          title="Personal Portfolio"
        >
          <img
            class="sc-gKXOVf fmLhIk"
            src="portfolio.png"
          />
        </div>
        <p
          class="sc-eCYdqJ hNHXnO"
        >
          Personal Portfolio. Using styled components and typescript for the first time
        </p>
        <ul
          class="sc-ftvSup fwQdZr"
        >
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="react.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="typescript.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="styled.svg"
            />
          </li>
          <li
            class="sc-papXJ"
          >
            <img
              alt=""
              class="sc-jqUVSM fvaYrv"
              src="jest.svg"
            />
          </li>
        </ul>
        <ul
          class="sc-iBkjds gIKFUO"
        >
          <li>
            <a
              href="https://franciscasibu.codes/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              href="https://github.com/nevz9/portfolio/"
              rel="noreferrer noopener"
              target="_blank"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="16 18 22 12 16 6"
                />
                <polyline
                  points="8 6 2 12 8 18"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
